lib/Sema/TypeChecker.cpp

```diff
@@ -691,6 +692,7 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
       TC.processREPLTopLevel(SF, TLC, StartElem);
 
     typeCheckFunctionsAndExternalDecls(TC);
+    performTopLevelDeclDiagnostics(SF);
```

Comment:
Nice! This also un-covered a lot of masked warnings since I shared the `VarDeclUsageChecker` across multiple statements. This produced a bunch of new UT failures that I fixed and also un-covered a bug when `guard` adds a variable to the global scope.